core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1187,6 +1192,20 @@ public TableMetadata build() {
           previousFiles, previousFileLocation, base.lastUpdatedMillis(), properties);
       List<HistoryEntry> newSnapshotLog = updateSnapshotLog(snapshotLog, snapshotsById, currentSnapshotId, changes);
 
+      if (refs.isEmpty() && currentSnapshotId != -1) {
```

Comment:
Will add tests for this if we deem setting main here is the right way to go

Reply:
+1